cram_pr2/cram_pr2_shopping_demo/src/plans.lisp

```diff
@@ -0,0 +1,173 @@
+                        (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+      
+      (setf map->obj (second (fourth (desig:desig-prop-value ?object :pose))))
+      (setf trans (cl-transforms:transform* (cl-transforms:transform-inv map->base) map->obj))
+
+      (setf ?newpose (cl-transforms-stamped:make-pose-stamped
+                      "base_footprint"
+                      0.0
+                      (cl-transforms:translation trans)
+                      (cl-transforms:orientation
+                       (second (first (desig:desig-prop-value ?object :pose))))))
+      
+      (setf ?newTransform (cl-transforms-stamped:make-transform-stamped
+                          "base_footprint"
+                          (desig:desig-prop-value ?object :name)
+                          0.0
+                          (cl-transforms:translation trans)
+                          (cl-transforms:rotation
+                           (second (second (desig:desig-prop-value ?object :pose))))))
+
+      ;; constructing the new Object designator with the new pose and transformation
+      (setf ?object (desig:copy-designator ?object
+                                           :new-description
+                                           `((:type ,(desig:desig-prop-value ?object :type))
+                                             (:name ,(desig:desig-prop-value ?object :name))
+                                             (:pose ((:pose ,?newpose)
+                                                     (:transform ,?newTransform)
+                                                     (:pose-in-map ,(second (third (desig:desig-prop-value ?object :pose))))
+                                                     (:transform-in-map ,(second (fourth (desig:desig-prop-value ?object :pose)))))))))
+
+     
+    ;; (print ?object)
+      
+
+      ;; selecting the grasping arm
+      (if (< (cl-transforms:y (cl-transforms:translation ?newtransform)) 0)
+          (setf ?grasping-arm :right)
+          (setf ?grasping-arm :left))
+
+      ;; if the object is to far left move to the left
+      (when (> (cl-transforms:y (cl-transforms:translation ?newtransform)) 0.8)
+        (setf ?grasping-arm :left)
+        (exe:perform (desig:a action (type going) (target (desig:a location (pose ?grasp-pose-left))))))
+      ;; if it is to far on the right move to the right
+      (when (< (cl-transforms:y (cl-transforms:translation ?newtransform)) -0.8)
+        (setf ?grasping-arm :right)
+        (exe:perform (desig:a action (type going) (target (desig:a location (pose ?grasp-pose-right))))))
+
+      (print ?grasping-arm)
+      
+      (exe:perform (desig:an action
+                             (type picking-up)
+                             (arm ?grasping-arm)
+                             (grasp left-side)
+                             (object ?object)))
+      
+      (exe:perform (desig:an action (type going) (target (desig:a location (pose ?pose-near-table)))))
```

Comment:
I would keep the formatting of all designators consistent with each other. Either use new lines everywhere or don't everywhere.